cpp/src/plasma/external_store.cc

```diff
@@ -0,0 +1,67 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <iostream>
+#include <sstream>
+
+#include "external_store.h"
+
+namespace plasma {
+
+std::shared_ptr<std::map<std::string, std::shared_ptr<ExternalStore>>> ExternalStores::external_stores_ = nullptr;
```

Comment:
Make this a `std::unordered_map` and also remove the `shared_ptr`. Then this map will be initialised on startup. There is no need for using a pointer indirection here.